### ISSUE TITLE
508-defect-4 [COGNITION, KEYBOARD]: 

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#43252]
# [508-defect-4](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-4)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** Josh

## User Story or Problem Statement

As a keyboard user interested in inputting my VA file number, I expect that I should be able to tab to the VA file number field without receiving an error.

## Details

Currently, the form provides error validation when users tab out of required fields. That presents a tricky scenario where there are _two_ options that transition between optional and required based on the user's intent and input. For example, in the user story above, a keyboard user may become confused that they received an error telling them that, in fact, their _social security number_ is the required field instead of their VA file number (people sometimes miss labels).

### Bandaid fix
- [ ] Error validation for VA file number and social security number does not occur until _after_ the user clicks the continue button

### Semantic fix
- [ ] One thing per page is used to chunk conditional questions on separate pages 

e.g.: 

Instead of one page with 6+ questions...

We could chunk those into separate, smaller pages:
Page 1: What is the deceased Veteran's name?
Page 2: Did they go by other names?
Page 3: [Input names]
Page 4: Do you want to input the deceased Veteran's social security number or their VA file number?
Page 5: Input the deceased Veteran's VA file number
Page 6: Where and when was the deceased Veteran born?

## Steps to Recreate

1. [Go to apply for burials page and start without signing in](https://www.va.gov/burials-and-memorials/application/530/veteran-information/burial)
2. Go to the Deceased Veteran information page
3. Tab past the SSN field and confirm an error is thrown

## Proposed Solution (if known)

Two solutions are provided in the acceptance criteria. The bandaid fix addresses the immediate problem at hand, but we should probably explore improving the form design of this flow to reduce bugs caused be workarounds like this one.

## WCAG or Vendor Guidance (optional)

1. [Better Form Design: One Thing Per Page (Case Study)](https://www.smashingmagazine.com/2017/05/better-form-design-one-thing-per-page/)
2. [Gov.uk one thing per page](https://designnotes.blog.gov.uk/2015/07/03/one-thing-per-page/)

## Screenshots or Trace Logs

https://user-images.githubusercontent.com/14154792/175332004-e130905a-dedb-4491-a059-1a649759b6dc.mov

**Comment by steele-lm** [comment]
Closing this ticket. The new version of the form on VA.gov addresses this issue.
